transport_test.go

```diff
@@ -125,6 +131,91 @@ func TestGetRequestBodyFromEventCompletelyInvalid(t *testing.T) {
 	}
 }
 
+func TestGetEnvelopeFromBody(t *testing.T) {
+	body := getRequestBodyFromEvent(&Event{
+		Type:           transactionType,
+		Spans:          []*Span{},
+		StartTimestamp: time.Unix(3, 0).UTC(),
+		Timestamp:      time.Unix(5, 0).UTC(),
+	})
+	env := getEnvelopeFromBody(body)
```

Comment:
Pick your favorite date to pass to this function with the new argument as in https://github.com/getsentry/sentry-go/pull/235/files#r434621423, then some 20 lines of code can be deleted. No need to parse envelopes :)